internal/cli/uninstall.go

```diff
@@ -55,24 +56,43 @@ func (c *UninstallCommand) Run(args []string) int {
 		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
 		return 1
 	}
-	c.ui.Output(
-		"Uninstalling Waypoint server with context %q",
-		contextDefault,
-		terminal.WithSuccessStyle(),
-	)
 
-	// Get the platform early so we can validate it.
-	p, ok := serverinstall.Platforms[strings.ToLower(c.platform)]
-	if !ok {
-		if c.platform == "" {
+	var ctxConfig *clicontext.Config
+	if contextDefault != "" {
+		ctxConfig, err = c.contextStorage.Load(contextDefault)
+		if err != nil {
+			c.ui.Output("Error loading context %q: %s", contextDefault, err.Error(), terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	serverPlatform := c.platform
+	if ctxConfig.Server.Platform == "" && c.platform != "" {
+		c.ui.Output(
+			"No platform set on server context. Will attempt to uninstall requested "+
+				"platform %q",
+			c.platform,
+			terminal.WithWarningStyle(),
+		)
+	} else if serverPlatform == "" {
+		// attempt to set the server platform so the platform flag isn't required.
+		serverPlatform = ctxConfig.Server.Platform
+
+		if serverPlatform == "" && c.platform == "" {
```

Comment:
At this point, we've already checked that `c.platform` is empty via the `else if` line checking if `serverPlatform` is empty, since we set `serverPlatform` to `c.platform` on line 69 and don't reset it until after we check if it's empty. Maybe consolidate some of our `c.platform` and `serverPlatform` references so it's clear what we're checking all the way through?
```suggestion
		if serverPlatform == "" {
```